File: ios/Stopwatch/core/use_cases/impl/LoggingUseCaseImpl.swift
class LoggingUseCaseImpl: LoggingUseCase {
    private let loggingService: LoggingService
    
    init(_ loggingService: LoggingService) {
        self.loggingService = loggingService
    }
    
    func debug(tag: String, message: String) {
        loggingService.debug(tag: tag, message: message)
    }
    
    func info(tag: String, message: String) {
        loggingService.info(tag: tag, message: message)
    }
    
    func warn(tag: String, message: String) {
        loggingService.warn(tag: tag, message: message)
    }
    
    func error(tag: String, message: String, error: (any Error)? = nil) {
        loggingService.error(tag: tag, message: message, error: error)
    }
}

File: android/app/src/main/java/com/rafaeltmbr/stopwatch/core/use_cases/impl/LoggingUseCaseImpl.kt
package com.rafaeltmbr.stopwatch.core.use_cases.impl

import com.rafaeltmbr.stopwatch.core.services.LoggingService
import com.rafaeltmbr.stopwatch.core.use_cases.LoggingUseCase

class LoggingUseCaseImpl(
    private val loggingService: LoggingService
) : LoggingUseCase {
    override fun debug(tag: String, message: String) = loggingService.debug(tag, message)
    override fun info(tag: String, message: String) = loggingService.info(tag, message)
    override fun warn(tag: String, message: String) = loggingService.warn(tag, message)
    override fun error(tag: String, message: String, exception: Exception?) =
        loggingService.error(tag, message, exception)
}